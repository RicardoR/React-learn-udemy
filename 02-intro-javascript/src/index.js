import { giphyRandomEndpoint, apiKey } from './data/giphy-data';

const getImagePromise = () => Promise.resolve('image with promise');

getImagePromise().then((image) => {
  console.log(image);
});

const getImageAsync = async () => {
  try {
    const fetchRequest = await fetch(
      `${giphyRandomEndpoint}?api_key=${apiKey}`
    );
    const { data } = await fetchRequest.json();
    return data.images.original.url;
  } catch (error) {
    console.error(error);
  }
};

getImageAsync()
  .then((imageUrl) => {
    const imageElement = document.createElement('img');
    imageElement.src = imageUrl;
    document.body.appendChild(imageElement);
  })
  .catch((error) => console.info(error));
