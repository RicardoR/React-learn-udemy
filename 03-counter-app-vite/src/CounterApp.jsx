 import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {

    const [counter, updateCounter] = useState(value);

    const updateValue = (val) => {
        updateCounter(() => counter + val )
    };

    return (
        <>
            <h1> CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={() => updateValue(1)}>+1</button>
            <button onClick={() => updateValue(2)}>+2</button>
            <button onClick={() => updateValue(-1)}>-1</button>
            <button onClick={() => updateValue(-2)}>-2</button>
        </>
    );
};


CounterApp.propTypes = {
    value: PropTypes.number.isRequired
};
