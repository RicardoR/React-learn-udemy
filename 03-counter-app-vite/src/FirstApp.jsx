import React from 'react';
import PropTypes from 'prop-types';

const getSum = (a,b) => {
    return a + b;
};

export const FirstApp = ({name, surname, age, status }) => {
    return (
        <>
            <h1> Name: { name } </h1>
            <p> Surname: { surname } </p>
            <span> Age: { age } years </span>
            <hr/>
            <div> Sum: { getSum(2,2) } </div>
            <div>Status: { status }</div>
        </>
    );
};


FirstApp.propTypes = {
    name: PropTypes.string.isRequired,
    surname: PropTypes.string.isRequired,
    age: PropTypes.number.isRequired,
    status: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    status: 'Living'
};
