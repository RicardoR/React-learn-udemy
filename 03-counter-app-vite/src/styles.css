html, body {
    background-color: #5c62c5;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 25px;
}
