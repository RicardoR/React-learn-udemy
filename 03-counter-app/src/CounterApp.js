import PropTypes from 'prop-types';
import { Fragment, useState } from 'react';

const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value);
  
  // By default setCaunter retrive the value from the state
  const decrease = () => setCounter((c) => c - 1); 
  const increase = () => setCounter(counter + 1);
  const reset = () => setCounter(value);

  return (
    <Fragment>
      <h1>CounterApp</h1>
      <button id='resetBtn' onClick={reset}>Reset</button>
      <div className="counter-controls">
        <button id='decreaseBtn' onClick={decrease}>- 1</button>
        <p> {counter} </p>
        <button id='increaseBtn' onClick={increase}>+ 1</button>
      </div>
    </Fragment>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
    value: 5,
}

export default CounterApp;