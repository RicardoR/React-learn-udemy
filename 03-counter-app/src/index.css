html, body {
    align-items: center;
    background-color: #21232a;
    color: white;
    display: flex;
    font-family: 'Roboto', sans-serif;
    font-size: 1.3rem;
    min-height: 100%;
    justify-content: center;
    flex-direction: column;
}

body > div {
    width: 100%;
    text-align: center;
}

button {
    background-color: blueviolet;
    border: none;
    border-radius: 3px;
    color: white;
    cursor: pointer;
    font-size: 0.8rem;
    padding: 0.2rem 0.3rem;
    transition: all 0.2s ease-in-out;
    margin-left: 8px;
    margin-right: 8px;
    cursor: pointer;
}

.counter-controls {
    margin: auto;
    display: flex;
    margin-bottom: 1rem;
    align-items: center;
    align-self: center;
    justify-content: center;
}