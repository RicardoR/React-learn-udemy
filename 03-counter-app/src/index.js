import React from 'react';
import ReactDOM from 'react-dom';
import RamonesApp from './RamonesApp';
import './index.css';

const rootDiv = document.querySelector('#root');
const ramones = {
  band: 'Ramones',
  origin: 'Los Angeles, California',
};
ReactDOM.render( <RamonesApp {...ramones} />, rootDiv);