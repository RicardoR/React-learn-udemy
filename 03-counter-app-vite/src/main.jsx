import React from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from "./FirstApp";
import "./styles.css";
import { CounterApp } from "./CounterApp";

const developer = {
    name: 'Ricardo',
    surname: 'Rodr√≠guez',
    age: 41
};

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWorldApp/>
        <FirstApp {...developer} />
        <CounterApp value={4}/>
    </React.StrictMode>
);
