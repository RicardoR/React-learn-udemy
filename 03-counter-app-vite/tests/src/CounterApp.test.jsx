import {render, screen} from "@testing-library/react";
import {CounterApp} from "../../src/CounterApp";

describe('CounterApp Ut', () => {
    const initialValue = 100;
    const {container} = render(<CounterApp value={initialValue}/>);

   test('should do match with the snapshot', () => {
       expect(container).toMatchSnapshot();
   });

   test('should display the initial value', async() => {
       await render(<CounterApp value={initialValue}/>);
       expect(screen.getByTestId('test-counter').innerHTML).toContain(`${initialValue}`);
   });
});
