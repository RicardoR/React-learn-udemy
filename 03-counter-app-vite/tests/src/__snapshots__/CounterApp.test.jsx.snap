// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp Ut should do match with the snapshot 1`] = `
<div>
  <h1>
     CounterApp
  </h1>
  <h2
    data-testid="test-counter"
  >
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button>
    +2
  </button>
  <button>
    -1
  </button>
  <button>
    -2
  </button>
  <button>
    Reset
  </button>
</div>
`;
